micrometer-core/src/main/java/io/micronaut/configuration/metrics/binder/web/HttpMeterFilterFactory.java

```diff
@@ -41,41 +41,74 @@
 @Requires(property = WebMetricsPublisher.ENABLED, notEquals = FALSE)
 public class HttpMeterFilterFactory {
 
+    public static final double SECONDS_TO_NANOS = 1_000_000_000d;
+
     /**
      * Configure new MeterFilter for http.server.requests metrics.
      *
      * @param percentiles The percentiles
+     * @param histogram If a histogram should be published
+     * @param min       the minimum time (in s) value expected.
+     * @param max       the maximum time (in s) value expected.
+     * @param slos      the user-defined service levels objectives (in s) to create.
      * @return A MeterFilter
      */
     @Bean
     @Singleton
-    @Requires(property = MICRONAUT_METRICS_BINDERS + ".web.server.percentiles")
-    MeterFilter addServerPercentileMeterFilter(@Value("${" + MICRONAUT_METRICS_BINDERS + ".web.server.percentiles}") Double[] percentiles) {
-        return getMeterFilter(percentiles, METRIC_HTTP_SERVER_REQUESTS);
+    @Requires(property = MICRONAUT_METRICS_BINDERS + ".web.server")
+    MeterFilter addServerPercentileMeterFilter(
+        @Value("${" + MICRONAUT_METRICS_BINDERS + ".web.server.percentiles:}") Double[] percentiles,
+        @Value("${" + MICRONAUT_METRICS_BINDERS + ".web.server.histogram:false}") Boolean histogram,
+        @Value("${" + MICRONAUT_METRICS_BINDERS + ".web.server.min:-1}") Double min,
+        @Value("${" + MICRONAUT_METRICS_BINDERS + ".web.server.max:-1}") Double max,
+        @Value("${" + MICRONAUT_METRICS_BINDERS + ".web.server.slos:}") Double[] slos
```

Comment:
generally with this many properties it would be good to decouple it into a `@ConfigurationProperties(MICRONAUT_METRICS_BINDERS + ".web.server)` java `record` or something since these generate configuration documentation and auto-completion in the IDE